From e2f7f15f184412cd28d578d05ba98e27339af743 Mon Sep 17 00:00:00 2001
From: Jack Tan <jack.tan@windriver.com>
Date: Wed, 5 Sep 2012 16:47:37 +0800
Subject: [PATCH 22/27] MIPS: octeon2: generate an assembly file of the device
 tree

Generate an assembly file to wrap the output of the device tree compiler

Based on SDK 2.3.0-427

Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/cavium-octeon/Makefile |   28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/arch/mips/cavium-octeon/Makefile b/arch/mips/cavium-octeon/Makefile
index bc96e29..12fcd5a 100644
--- a/arch/mips/cavium-octeon/Makefile
+++ b/arch/mips/cavium-octeon/Makefile
@@ -24,8 +24,32 @@ DTB_FILES = $(patsubst %.dts, %.dtb, $(DTS_FILES))
 
 obj-y += $(patsubst %.dts, %.dtb.o, $(DTS_FILES))
 
-$(obj)/%.dtb: $(src)/%.dts FORCE
-	$(call if_changed_dep,dtc)
+# DTC
+# ---------------------------------------------------------------------------
+
+# Generate an assembly file to wrap the output of the device tree compiler
+quiet_cmd_dt_S_dtb= DTB    $@
+cmd_dt_S_dtb=                                           \
+(                                                       \
+        echo '.section .dtb.init.rodata,"a"';           \
+        echo '.balign 8';               \
+        echo '.global __dtb_$(*F)_begin';               \
+        echo '__dtb_$(*F)_begin:';                      \
+        echo '.incbin "$<" ';                           \
+        echo '__dtb_$(*F)_end:';                        \
+        echo '.global __dtb_$(*F)_end';                 \
+        echo '.balign 8';               \
+) > $@
+
+$(obj)/%.dtb.S: $(obj)/%.dtb
+	$(call cmd,dt_S_dtb)
+
+quiet_cmd_dtc = DTC     $@
+cmd_dtc = $(objtree)/scripts/dtc/dtc -O dtb -o $@ -b 0 $(DTC_FLAGS) $<
+
+
+$(obj)/%.dtb: $(src)/%.dts
+	$(call cmd,dtc)
 
 # Let's keep the .dtb files around in case we want to look at them.
 .SECONDARY:  $(addprefix $(obj)/, $(DTB_FILES))
-- 
1.7.9.7

